### PR TITLE
Transaction validity and to-plain conversion for PoW transactions

### PR DESCRIPTION
**Pull request by sisou**
Ensure PoW transactions can be properly converted to their plain representation, which is primarily done during address transaction history fetching in the web-client. Before this change, PoW transactions could not be returned and the process panicked, due to different serializations of contract creation data, HTLC proofs and signature proofs.

I have tried to document all changes and differences in code comments, let me know if you require more info, here or in the code.

## Pull request checklist

- [x] All tests pass. The project builds and runs.
- [x] I have resolved any merge conflicts.
- [x] I have resolved all `clippy` and `rustfmt` warnings.

### CODE REVIEW

File: pow-migration/Cargo.toml [hunk 1]
```diff
@@ -72,6 +72,7 @@ build-data = "0.2"
 
 [dev-dependencies]
 serde_json = "1.0"
+nimiq-web-client = { workspace = true, default-features = false, features = ["primitives"] }
 
 nimiq-test-log = { workspace = true }
```
Comment (jsdanielh) [inline pow-migration/Cargo.toml]:
```suggestion

nimiq-test-log = { workspace = true }
nimiq-web-client = { workspace = true, default-features = false, features = ["primitives"] }
```

File: primitives/transaction/src/account/htlc_contract.rs [hunk 2]
```diff
@@ -225,6 +243,49 @@ impl CreationTransactionData {
     }
 }
 
+#[derive(Clone, Debug, Default, Deserialize, Serialize)]
+pub struct PoWCreationTransactionData {
```
Comment (jsdanielh) [inline primitives/transaction/src/account/htlc_contract.rs:260]: Could you add some `rustdoc`?

File: primitives/transaction/src/account/htlc_contract.rs [hunk 3]
```diff
@@ -225,6 +243,49 @@ impl CreationTransactionData {
...
+    }
+
+    pub fn verify(&self) -> Result<(), TransactionError> {
+        if self.hash_count == 0 {
+            return Err(TransactionError::InvalidData);
+        }
+        Ok(())
+    }
+
+    pub fn into_pos(self, genesis_number: u32, genesis_timestamp: u64) -> CreationTransactionData {
```
Comment (jsdanielh) [inline primitives/transaction/src/account/htlc_contract.rs]:
```suggestion
    pub fn into_pos(self, genesis_block_number: u32, genesis_timestamp: u64) -> CreationTransactionData {
```

File: primitives/transaction/src/account/htlc_contract.rs [hunk 4]
```diff
@@ -340,6 +401,129 @@ impl OutgoingHTLCTransactionProof {
     }
 }
 
+#[derive(Clone, Debug)]
+pub struct PoWRegularTransfer {
```
Comment (jsdanielh) [inline primitives/transaction/src/account/htlc_contract.rs:435]: Could you add some `rustdoc`?

File: primitives/transaction/src/account/vesting_contract.rs [hunk 5]
```diff
@@ -198,3 +206,120 @@ impl CreationTransactionData {
...
+                    start_block,
+                    step_blocks,
+                    step_amount,
+                    total_amount,
+                }
+            }
+            _ => return Err(TransactionError::InvalidData),
+        })
+    }
+    pub fn parse(transaction: &Transaction) -> Result<Self, TransactionError> {
```
Comment (jsdanielh) [inline primitives/transaction/src/account/vesting_contract.rs]:
```suggestion
    }

    pub fn parse(transaction: &Transaction) -> Result<Self, TransactionError> {
```

File: primitives/transaction/src/account/vesting_contract.rs [hunk 6]
```diff
@@ -198,3 +206,120 @@ impl CreationTransactionData {
...
+        PoWCreationTransactionData::parse_data(&transaction.recipient_data, transaction.value)
+    }
+
+    pub fn into_pos(self, genesis_number: u32, genesis_timestamp: u64) -> CreationTransactionData {
```
Comment (jsdanielh) [inline primitives/transaction/src/account/vesting_contract.rs]:
```suggestion
    pub fn into_pos(self, genesis_block_number: u32, genesis_timestamp: u64) -> CreationTransactionData {
```

File: primitives/transaction/src/signature_proof.rs [hunk 7]
```diff
@@ -350,6 +350,42 @@ impl WebauthnExtraFields {
     }
 }
 
+#[derive(Clone, Debug, Deserialize)]
+pub struct PoWSignatureProof {
```
Comment (jsdanielh) [inline primitives/transaction/src/signature_proof.rs:357]: Could you add some `rustdoc`?

**Comment by sisou** [comment]
Ready for review.

File: pow-migration/Cargo.toml [hunk 8]
```diff
@@ -74,6 +74,7 @@ build-data = "0.2"
 serde_json = "1.0"
 
 nimiq-test-log = { workspace = true }
+nimiq-web-client = { workspace = true, default-features = false, features = ["primitives"] }
```
Comment (jsdanielh) [inline pow-migration/Cargo.toml]: The  `default-features = false` is inherited from the workspace
```suggestion
nimiq-web-client = { workspace = true, features = ["primitives"] }
```

**Review by jsdanielh — APPROVED** [review]

**Pull request merged**